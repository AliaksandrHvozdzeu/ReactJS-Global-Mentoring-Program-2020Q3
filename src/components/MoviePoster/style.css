.poster {
    width: 250px;
    margin-left: -65px;
    margin-top: 10px;
    height: 350px;
}
