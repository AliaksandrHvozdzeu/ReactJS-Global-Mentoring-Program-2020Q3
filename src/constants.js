module.exports = Object.freeze({
  DEFAULT: [
    { label: 'Comedy', value: 'Comedy' },
    { label: 'Documentary', value: 'Documentary' },
    { label: 'Horror', value: 'Horror' },
    { label: 'Crime', value: 'Crime' },
  ],
  FILTER_DEFAULT : [
    {label: 'release', value: 'release'},
    {label: 'runtime', value: 'runtime'}
  ],
  MAPPING: {
    'release date': 'release',
    runtime: 'runtime'
  },
  FILTERS: ['All', 'Documentary', 'Comedy', 'Horror', 'Crime'],
  SORTING: ['Release', 'Runtime'],
  NOT_FOUND_POSTER: '/images/not-found.jpg',
  ARROWS: ['/images/arrow_up.png', '/images/arrow_down.png'],
  PASSWORD_HANDLE: ['/images/eye-hide.png', '/images/eye-show.png']
});
