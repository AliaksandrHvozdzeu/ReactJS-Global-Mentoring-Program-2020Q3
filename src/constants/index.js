module.exports = Object.freeze({
  DEFAULT: [
    { label: 'Comedy', value: 'Comedy' },
    { label: 'Documentary', value: 'Documentary' },
    { label: 'Horror', value: 'Horror' },
    { label: 'Crime', value: 'Crime' },
  ],
  FILTER_DEFAULT: [
    { label: 'release', value: 'release' },
    { label: 'runtime', value: 'runtime' },
  ],
  MAPPING: {
    'release date': 'release',
    runtime: 'runtime',
  },
  FILTERS: ['All', 'Documentary', 'Comedy', 'Horror', 'Crime'],
  SORTING: ['Release', 'Runtime', 'Budget', 'Revenue', 'Vote average', 'Vote count'],
  NOT_FOUND_POSTER: '/images/not-found.png',
  ARROWS: ['/images/arrow_up.png', '/images/arrow_down.png'],
  PASSWORD_HANDLE: ['/images/eye-hide.png', '/images/eye-show.png'],
  DEFAULT_COUNT_FILTER_ITEM: 6,
  ALL_GENRES: 'All',
  CLEAR_SEARCH_INPUT_IMAGE: '/images/close.png',
  EDIT_SUCCESS: 'edit-success',
  DELETE_SUCCESS: 'delete-success',
  ADD_SUCCESS: 'add-success',
  ERROR: 'error',
  FILTER_GENRE_TYPE: 'filter/genre',
  FILTER_ORDER_TYPE: 'filter/order',
  FETCH_MOVIES_LIST_SUCCESS_TYPE: 'movies/fetchMoviesSuccess',
  LOAD_MOVIES_DEFAULT_TYPE: 'movies/loadMovieDefault',
  DELETE_MOVIES_SUCCESS_TYPE: 'movies/deleteMovie',
  EDIT_MOVIE_SUCCESS_TYPE: 'movies/editMovieSuccess',
  LOAD_MOVIES_BY_TITLE_TYPE: 'movies/loadMovieByTitle',
  ADD_MOVIE_SUCCESS_TYPE: 'movies/addMovieSuccess',
  SET_FILTERED_RESULTS_TYPE: 'movies/filteredResults',
  CLEAR_SEARCH_INPUT_TYPE: 'movies/clearSearch',
  PREVIEW_MOVIE_TYPE: 'movies/preview',
  UPDATE_GENRES_TYPE: 'genres/updateGenres',
  LOADER_TYPE: 'common/loader',
  ERROR_TYPE: 'common/error',
  RESET_TYPE: 'common/reset',
  PUT_METHOD: 'put',
  POST_METHOD: 'post',
  DELETE_METHOD: 'delete',
  SHOW_SUCCESS_MESSAGE: 'common/successMessage',
  ARROW_IMAGES_ARRAY: [{
    src: './images/scroll_arrow_down.png',
    alt: 'scroll button down',
  }, {
    src: './images/scroll_arrow_up.png',
    alt: 'scroll button up',
  }],
  FORMIK_ERRORS: [{
    title: 'Title is required field',
    release_date: 'Release date is required field',
    poster_path: 'Poster should be a valid url',
    genre: 'Select at least one genre',
    overview: 'Film description is required',
    runtime: 'Runtime should be numeric value more then 0'
  }],
  DEFAULT_DATE_FORMAT: 'MM/dd/yyyy',
});
