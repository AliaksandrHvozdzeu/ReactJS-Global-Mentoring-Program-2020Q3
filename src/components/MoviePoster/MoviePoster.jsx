import React from 'react';
import PropTypes from 'prop-types';
import NoPoster from '../MovieNoPoster';
import './style.css';

function showPoster(src, alt) {
  if (!src) {
    return <NoPoster/>;
  }
  return <img className="poster" src={src} alt={alt}/>;
}

const MoviePoster = ({ src, alt }) => (
  showPoster(src, alt)
);

MoviePoster.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string,
  width: PropTypes.number
};

export default MoviePoster;
