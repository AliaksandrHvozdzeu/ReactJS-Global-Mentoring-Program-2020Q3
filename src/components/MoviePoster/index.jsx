import React from 'react';
import PropTypes from 'prop-types';
import './style.css';
import NoPoster from '../MovieNoPoster';

function showPoster(src, alt) {
  if (!src) {
    return <NoPoster/>;
  }
  return <img className="poster" src={src} alt={alt}/>;
}

const MoviePoster = ({ src, alt }) => (
  showPoster(src, alt)
);

MoviePoster.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string,
  width: PropTypes.number
};

export default MoviePoster;
