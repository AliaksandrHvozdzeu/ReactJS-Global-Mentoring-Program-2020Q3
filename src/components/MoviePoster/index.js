import MoviePoster from './js/MoviePoster';

export default MoviePoster;
