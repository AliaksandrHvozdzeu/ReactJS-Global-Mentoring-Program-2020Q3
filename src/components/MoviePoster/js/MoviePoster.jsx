// eslint-disable-next-line no-unused-vars
import React from 'react';
import PropTypes from 'prop-types';
import UseDefaultPoster from '../../../hooks/useDefaultPoster';
import Constants from '../../../constants';
import '../css/MoviePoster.css';

export default function MoviePoster({ src, alt, className, fallback, onClick, onKeyDown }) {
  return UseDefaultPoster(src, alt, className, fallback, onClick, onKeyDown);
}

MoviePoster.propTypes = {
  src: PropTypes.string.isRequired,
  fallback: PropTypes.string.isRequired,
  alt: PropTypes.string,
  className: PropTypes.string,
};

MoviePoster.defaultProps = {
  src: Constants.DEFAULT_POSTER,
  fallback: Constants.DEFAULT_POSTER,
  alt: 'movie poster logo',
  className: 'poster',
};
