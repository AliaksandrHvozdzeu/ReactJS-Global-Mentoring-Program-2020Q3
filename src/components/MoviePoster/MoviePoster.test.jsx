import React from "react";
import rerender from "react-test-renderer";
import MoviePoster from "./js/MoviePoster";
import { jest } from "@jest/globals";
import * as Constants from "../../constants";

describe("when MoviePoster", () => {

  const src = "https://upload.wikimedia.org/wikipedia/commons/4/47/PNG_transparency_demonstration_1.png";
  const alt = "image";
  const className = "movie-logo";
  const fallback = Constants.DEFAULT_POSTER;
  const onClick = jest.fn();
  const onKeyDown = jest.fn();

  test("then snapshot created", () => {
    const component = rerender.create(<MoviePoster src={src}
                                                   alt={alt}
                                                   className={className}
                                                   fallback={fallback}
                                                   onClick={onClick}
                                                   onKeyDown={onKeyDown} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
