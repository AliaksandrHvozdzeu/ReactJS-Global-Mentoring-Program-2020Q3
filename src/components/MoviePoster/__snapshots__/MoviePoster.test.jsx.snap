// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when MoviePoster then snapshot created 1`] = `
<img
  alt="image"
  className="movie-logo"
  onClick={[MockFunction]}
  onError={[Function]}
  onKeyDown={[MockFunction]}
  role="presentation"
  src="https://upload.wikimedia.org/wikipedia/commons/4/47/PNG_transparency_demonstration_1.png"
/>
`;
